atdgen/test/melange/melangespec.atd

```diff
@@ -108,3 +108,15 @@ type variant2 = [
   | A
   | C
 ] <ocaml repr="classic">
+
+type with_open_enum = [
```

Comment:
In the OCaml side, there's a test with a list + enum inside: 

https://github.com/ahrefs/atd/blob/77a07ed1f184f370d95ff882fa3a31911eecdf0f/atdgen/test/test3j.atd#L104

Do you think it's worth adding it to the melange suite as well?

Reply:
done